main.c

```diff
@@ -0,0 +1,203 @@
+ *
+ */
+
+char* modename(int mode)
+{
+	switch (mode) {
+		case 1: return "quiet batch mode"; 
+		case 2: return "verbose batch mode";
+		case 3: return "interactive mode";
+		default: break;
+	}
+	return "quiet batch mode";
+}
+
+int main(int argc, char **argv)
+{
+	// filenames up to 400 char
+	char* dict_name = malloc(401 * sizeof(char*));
+	char* file_name = malloc(401 * sizeof(char*));
+	char* save_file = malloc(401 * sizeof(char*));
+
+	// default dict name
+	strcpy(dict_name,"tests/sample_dict.txt");
+
+	/*
+		1: quiet batch
+		2: verbose batch
+		3: interactive
+	*/
+	int mode = 3;
+
+	/* Parse Command Line Args */
+	char c;
+
+	/* If command line contains just the file at argv[1], write it into file_name */
+	if (fileexists(argv[1])) strcpy(file_name,argv[1]);
+
+
+    // Parse the initial command line and 
+	while ((c = getopt(argc,argv,"d:i:v:s:q:")) != -1) {
+		switch (c) {
+		case 'd':
+			if (!fileexists(optarg)) {  // this checks if the file actually exists
+				shell_error("Dictionary input file path invalid");
+				return EXIT_FAILURE;
+			}
+			strcpy(dict_name,optarg);
+			if (mode == 3) shell_input(optarg, "dictionary");
+			break;
+		case 'i':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 3;
+            strcpy(file_name,optarg);
+			// if file name is not valid, print error
+			break;
+		case 'v':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 2;
+            strcpy(file_name,optarg);
+			break;
+		case 'q':
+			if (!fileexists(optarg)) {
+				shell_error("Input file path invalid");
+				return EXIT_FAILURE;
+			}
+			mode = 1;
+			strcpy(file_name,optarg);
+			break;
+		case 's':
+            if (strstr(optarg,".txt\0") == NULL) {    // does not save to a *.txt file
+                  shell_error("Input save file path invalid");
+                  return EXIT_FAILURE;
+            }
+			strcpy(save_file,optarg);
+            if (mode == 3) shell_input(optarg,"file save destination");
+			break;
+		default:
+			shell_usage();
+            exit(0);
+		}
+	}
+
+	int *quit = malloc(sizeof(int*)); // this enables return to main page, should user choose it at save_page
+	*quit = 0;
+
+  while (!(*quit)) {
+	if (fileexists(file_name)) {	// if file exists, then bypass main page
+		*quit = 1;
+	}
+
+	/* main page: activated if there is no file to be parsed.
+	   can open help page, quit, or load filename / dictname */
+	main_page(quit, &mode, file_name, dict_name);
+	if ((*quit) == 2) { // user selected "quit" in main_page
+		shell_outro();
+		return 0;
+	}
+
+	/*
+		Initialize dictionary, declare names of files to be used
+	*/
+	dict_t* dict = dict_new();
+	if (dict_read(dict, dict_name) == EXIT_FAILURE) {
+		printf("Trouble reading dictionary, exiting program\n");
+        exit(0);
+	}
+
+	/*
+		Starting to Parse file!
+	 */
+	char* md = modename(mode);
+
+    if (mode == 3) {
```

Comment:
This is a magic number. Please `#define` constants so that we can do things like `if (mode == MODE_INTERACTIVE)` instead of `3`. It shouldn't be work to figure out what `3` means in this case.